### DISCUSSION TITLE
channel.flow vs connection.blocked

### DISCUSSION THREAD

**Discussion by gomoripeti** [rabbitmq/rabbitmq-server#5643] [Questions] [answered]
Does the RabbitMQ broker send `channel.flow` messages? Searching in the repo I only see that the amqp Erlang client supports receiving such message, and `rabbit_reader` also has clauses for it to receive from the client (though flow=false from the client is not supported any more) 
But I cannot find any place where the server would send it to a client.
https://github.com/rabbitmq/rabbitmq-server/search?l=Erlang&q=%22channel.flow%22

Otoh the server does send `connection.blocked` messages, however the amqp Erlang client apparently does not itself do anything about it, but leaves it to the user (by `register_blocked_handler`).

Asking because the edoc of `amqp_channel:call/3` says [here](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/amqp_client/src/amqp_channel.erl#L149-L150) that 
> This will return the atom 'blocked' if the server has
> throttled the client for flow control reasons.

However the atom `blocked` is only returned if the server sent a `channel.flow` message and not in case of a `connection.blocked` message (unless I'm missing something).

What is the current status? Which message should be used by the server and handled by the client?

**Comment by michaelklishin** [comment, marked as answer]
`channel.flow` was a part of AMQP 0-8-0 and was quickly deprecated and dropped as not particularly effective.

`connection.blocked` and `connection.unblocked` were introduced in 2013 as AMQP 0-9-1 extensions on top of TCP-based back pressure mechanism.

**Reply by gomoripeti** [reply]
thanks for the historic insight
